### ISSUE TITLE
Not working on plasma 5.21

### ISSUE THREAD

**Issue by erosostolato** [Flupp/sticky-window-snapping#11]
It was working before, but after I update it to version 5.21, it stopped working. I already tried to reinstall, change settings, but nothing works.  Is there a way I can debug or get any kind of log to find out what's going on?

**Comment by Flupp** [comment]
Do the following files and directories exist?
```
~/.local/share/kwin/scripts/sticky-window-snapping/
~/.local/share/kservices5/kwin-script-sticky-window-snapping.desktop
```

```
% tree ~/.local/share/kwin/scripts/sticky-window-snapping/
├── contents/
│   ├── code/
│   │   └── main.js
│   ├── config/
│   │   └── main.xml
│   └── ui/
│       └── config.ui
├── gpl-2.0.txt
└── metadata.desktop
```

You can start KWin with log output enabled:
https://github.com/Flupp/sticky-window-snapping/blob/a65c31e137aa8e85c6dc06204394ef583acc7510/kwin-with-scripting-logging.bash#L1-L3
Maybe you find some helpful output. (You can adapt the `QT_LOGGING_RULES` to your needs.)

If you are not sure if the script is actually loaded, try adding a debug print at the beginning of `~/.local/share/kwin/scripts/sticky-window-snapping/contents/code/main.js` and restart KWin as shown above.
```js
print('Hello World!');
```

**Comment by erosostolato** [comment]
This my folder structure:
```
% tree ~/.local/share/kwin/scripts/sticky-window-snapping/
├── contents
│   ├── code
│   │   ├── main.js
│   │   └── old.js
│   ├── config
│   │   └── main.xml
│   └── ui
│       └── config.ui
├── gpl-2.0.txt
└── metadata.desktop
```

I restarted kwin and got this message:
```
$ QT_LOGGING_RULES='*=false; kwin_scripting=true' exec kwin_x11 --replace 
OpenGL vendor string:                   AMD
OpenGL renderer string:                 Radeon RX 580 Series (POLARIS10, DRM 3.35.0, 5.4.0-66-generic, LLVM 11.0.1)
OpenGL version string:                  4.6 (Compatibility Profile) Mesa 20.3.4 - kisak-mesa PPA
OpenGL shading language version string: 4.60
Driver:                                 RadeonSI
GPU class:                              Arctic Islands
OpenGL version:                         4.6
GLSL version:                           4.60
Mesa version:                           20.3.4
X server version:                       1.20.9
Linux kernel version:                   5.4
Requires strict binding:                yes
GLSL shaders:                           yes
Texture NPOT support:                   yes
Virtual Machine:                        no
kwin_scripting: /home/rosostolato/.local/share/kwin/scripts/sticky-window-snapping/contents/code/main.js:38: error: enabledCurrently is not defined
```

**Comment by erosostolato** [comment]
So looking at the code, I saw that the `enabledCurrently` variable was not defined inside the code.
So I changed it from
`enabledCurrently = config.enabledUsually;`
to:
`var enabledCurrently = config.enabledUsually;`

It's not throwing error anymore but it still doesn't work... probably that variable should be global for kwin scripts, I'm not sure.

**Comment by Flupp** [comment]
Hmm. Strange. My KWin (5.21.1) does not complain about the missing `var`. Anyways, some more `var`s were already added in the current development version. Can you try if it works with that version?

https://github.com/Flupp/sticky-window-snapping/blob/master/contents/code/main.js

**Comment by erosostolato** [comment]
It did not work... I believe it has something related to my kde installation.

What happened was that I installed kde neon developer edition a couple of months ago but then when plasma 5.21.1 came out, I ran a script to switch back to the user edition. It seems that it worked but I see that I still have the beta version of kwin:

![image](https://user-images.githubusercontent.com/75269233/109392260-f1920300-78f9-11eb-9824-2a405348ceed.png)
![image](https://user-images.githubusercontent.com/75269233/109392277-02db0f80-78fa-11eb-9997-ec10d7620cec.png)

But it's weird that while I was using developer edition, sticky window was working just fine and it stopped working after I changed it, but it seems that kwin kept the same version but stopped working.

Maybe I'll need to reinstall kde neon =(
